drakrun/drakrun/draksetup.py

```diff
@@ -75,6 +76,79 @@ def ensure_zfs(ctx, param, value):
     return value
 
 
+def sanity_check():
+    logging.info("Checking xen-detect...")
+    proc = subprocess.run('xen-detect -N', shell=True)
+
+    if proc.returncode != 1:
+        logging.error('It looks like the system is not running on Xen. Please reboot your machine into Xen hypervisor.')
+        return
+
+    logging.info("Testing if xl tool is sane...")
+
+    try:
+        subprocess.check_output('xl info', shell=True, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        logging.exception("Failed to test xl command.")
+        return
+
+    if not perform_xtf():
+        logging.error("Your Xen installation doesn\'t pass the necessary tests.")
+        return
+
+
+def perform_xtf():
+    logging.info('Testing your Xen installation...')
+    module_dir = os.path.dirname(os.path.realpath(__file__))
+    cfg_path = os.path.join(module_dir, "tools", "test-hvm64-example.cfg")
+    firmware_path = os.path.join(module_dir, "tools", "test-hvm64-example")
+
+    with open(cfg_path, 'r') as f:
+        test_cfg = f.read().replace('{{ FIRMWARE_PATH }}', firmware_path).encode('utf-8')
+
+    with tempfile.NamedTemporaryFile() as tmpf:
+        tmpf.write(test_cfg)
+        tmpf.flush()
+
+        try:
+            subprocess.check_output('xl destroy test-hvm64-example', shell=True, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError:
+            pass
+
+        subprocess.check_output(f'xl create -p {tmpf.name}', shell=True, stderr=subprocess.STDOUT, timeout=30)
+
+        module_dir = os.path.dirname(os.path.realpath(__file__))
+        test_altp2m_tool = os.path.join(module_dir, "tools", "test-altp2m")
+
+        try:
+            output = subprocess.check_output([test_altp2m_tool, 'test-hvm64-example'], stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf-8', 'replace')
+            logging.error(f'Failed to enable altp2m on domain. Your hardware might not support Extended Page Tables. Logs:\n{output}')
+            return False
+
+        p = subprocess.Popen(['xl', 'console', 'test-hvm64-example'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        subprocess.check_output('xl unpause test-hvm64-example', shell=True, stderr=subprocess.STDOUT, timeout=30)
+        stdout_b, _ = p.communicate(timeout=10)
+
+        stdout_text = stdout_b.decode('utf-8')
+        stdout = [line.strip() for line in stdout_text.split('\n')]
+
+        for line in stdout:
+            if line == 'Test result: SUCCESS':
+                logging.info('Preflight check with Xen Test Framework: PASS')
+                return True
+
+    logging.error('Preflight check with Xen Test Framework doesn\'t pass. Your hardware might not support VT-x. Logs: \n{stdout_text}')
+    return False
+
+
+@click.command(help='Perform self-test to check Xen installation')
+def selftest():
+    if not sanity_check():
+        sys.exit(1)
```

Comment:
How can we test against `sanity_check` returning truthy value? It seems to me that it would always return `None`

Reply:
fixed, this was a leftover from refactor